Haptics/app/src/main/java/com/example/android/haptics/samples/ui/bounce/BounceRoute.kt

```diff
@@ -0,0 +1,327 @@
+            ) {
+                Ball(
+                    offsetY = transitionData.ballOffsetY,
+                    displayTouchIndicator = transitionData.isAtStart
+                )
+                Floor()
+            }
+        }
+    }
+}
+
+@Composable
+private fun Ball(offsetY: Dp = 0.dp, displayTouchIndicator: Boolean = false) {
+    Box(
+        modifier = Modifier
+            .offset(y = offsetY)
+            .size(BALL_SIZE)
+            .clip(CircleShape)
+            .background(MaterialTheme.colors.primaryVariant)
+    ) {
+        if (displayTouchIndicator) {
+            Icon(
+                Icons.Outlined.TouchApp, contentDescription = null,
+                Modifier.align(
+                    Alignment.Center
+                ),
+                MaterialTheme.colors.onPrimary
+            )
+        }
+    }
+}
+
+/**
+ * Floor the ball impacts against.
+ */
+@Composable
+private fun Floor() {
+    Box(
+        Modifier
+            .height(FLOOR_SIZE)
+            .fillMaxWidth()
+            .background(MaterialTheme.colors.primaryVariant)
+    )
+}
+
+/**
+ * Hold the transition values for the ball being dropped and reset.
+ */
+private data class TransitionData(
+    val ballOffsetY: Dp,
+    val isAtStart: Boolean, // Transition is complete and ball is in start position.
+    val isAtEnd: Boolean, // Transition is complete and now at rest on floor.
+    val isResetting: Boolean, // Currently being reset (ball raising back up) to starting state.
+    val isBouncing: Boolean, // Ball has been released and is bouncing.
+)
+/**
+ * Create a transition and return animation values for animating the ball drop.
+ */
+@Composable
+private fun updateTransitionData(ballState: BallState): TransitionData {
+    val transition =
+        updateTransition(ballState, label = "Transition between ball raised and dropped.")
+    val ballOffsetY by transition.animateDp(
+        transitionSpec = {
+            when {
+                BallState.Start isTransitioningTo BallState.End ->
+                    tween(DROP_ANIMATION_TIME_MS, easing = BounceEasing())
+                else ->
+                    tween(RESET_ANIMATION_TIME_MS, easing = LinearOutSlowInEasing)
+            }
+        }, label = "Ball drop y offset."
+    ) { state ->
+        when (state) {
+            BallState.Start -> BALL_START_POSITION
+            BallState.End -> BALL_END_POSITION
+        }
+    }
+    val isAtTargetState = transition.currentState === transition.targetState
+    return TransitionData(
+        ballOffsetY = ballOffsetY,
+        isAtStart = isAtTargetState && transition.currentState == BallState.Start,
+        isAtEnd = isAtTargetState && transition.currentState == BallState.End,
+        isBouncing = !isAtTargetState && transition.targetState == BallState.End,
+        isResetting = transition.currentState != BallState.Start && transition.targetState == BallState.Start,
```

Comment:
why not !isAtTargetState && transition.targetState == BallState.Start, for consistency with isBouncing?